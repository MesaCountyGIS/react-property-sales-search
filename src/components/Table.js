import React, { Component } from 'react';

  class Table extends Component {

      constructor(props){
          super(props)
          this.state = {
              isShown: false
          }
      }
    render() {
      return (
          <table style={{"display": this.state.isShown?"block": "none"}} id="table" class="row-border hover">
              <thead>
                  <tr>
                      <th>Parcel Number (Click for map)</th>
                      <th>Account Number (Click for details)</th>
                      <th>Reception No./Book Page (Click for Doc.)</th>
                      <th>Address</th>
                      <th>Sale Price</th>
                      <th>Sale Date</th>
                      <th>Grantee</th>
                      <th>Grantor</th>
                      <th>Actual Value</th>
                      <th>Qualification Code</th>
                      <th>Vacant Code</th>
                      <th>Neighborhood</th>
                      <th>Economic Area</th>
                      <th>Property Use</th>
                      <th>Total Heated Sq Ft</th>
                      <th>Price by Heated SQFT</th>
                      <th>Aprox. Acres</th>
                      <th>Architectual Style 1st Building</th>
                      <th>Total # Buildings</th>
                      <th>Earliest Effective Year Built</th>
                  </tr>
              </thead>
              <tbody></tbody>
          </table>
      );
    }
  }


export default Table;
