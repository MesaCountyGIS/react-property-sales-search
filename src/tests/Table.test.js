import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { shallow, mount, render } from 'enzyme';
import {stubConsoleError} from './utils/test-utils';
import Table from '../components/Table';

describe("<Table />", ()=>{
    stubConsoleError();

    test('renders without crashing', () => {
        const tree = renderer.create(
            <Table />
                ).toJSON();

      sinon.assert.notCalled(console.error)
      expect(tree).toMatchSnapshot();
    });

    `Add test for isOpen state. The Table should open when data is received`

});
