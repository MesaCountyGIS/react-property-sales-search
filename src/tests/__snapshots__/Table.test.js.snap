// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders without crashing 1`] = `
<table
  className="row-border hover"
  id="table"
  style={
    Object {
      "display": "none",
    }
  }
>
  <thead>
    <tr>
      <th>
        Parcel Number (Click for map)
      </th>
      <th>
        Account Number (Click for details)
      </th>
      <th>
        Reception No./Book Page (Click for Doc.)
      </th>
      <th>
        Address
      </th>
      <th>
        Sale Price
      </th>
      <th>
        Sale Date
      </th>
      <th>
        Grantee
      </th>
      <th>
        Grantor
      </th>
      <th>
        Actual Value
      </th>
      <th>
        Qualification Code
      </th>
      <th>
        Vacant Code
      </th>
      <th>
        Neighborhood
      </th>
      <th>
        Economic Area
      </th>
      <th>
        Property Use
      </th>
      <th>
        Total Heated Sq Ft
      </th>
      <th>
        Price by Heated SQFT
      </th>
      <th>
        Aprox. Acres
      </th>
      <th>
        Architectual Style 1st Building
      </th>
      <th>
        Total # Buildings
      </th>
      <th>
        Earliest Effective Year Built
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
